app/assets/javascripts/discourse/tests/integration/widgets/post-menu-test.js

```diff
@@ -39,19 +45,66 @@ discourseModule(
       },
     });
 
-    componentTest("remove extra button", {
+    componentTest("removes button based on callback", {
       template: hbs`{{mount-widget widget="post-menu" args=args}}`,
+
       beforeEach() {
-        this.set("args", {});
-        withPluginApi("0.8", (api) => {
-          api.removePostMenuButton("coffee");
+        this.set("args", { canCreatePost: true, canRemoveReply: true });
+
+        withPluginApi("0.14.0", (api) => {
+          api.removePostMenuButton("reply", (attrs) => {
+            return attrs.canRemoveReply;
+          });
         });
       },
+
+      afterEach() {
+        resetPostMenuExtraButtons();
+      },
+
       async test(assert) {
-        assert.ok(
-          !exists(".actions .extra-buttons .hot-coffee"),
```

Comment:
This would end up being a false positive previously if the test above wasn't run first and leaked its state.